### DISCUSSION TITLE
Airtable realtime continuous data sync

### DISCUSSION THREAD

**Discussion by Olmatro** [AntlerVC/firetable#1135] [Support / Q&A] [answered]
source: https://www.rowy.io/pricing
references : https://www.youtube.com/watch?v=zZxK9m2O_LI , https://roadmap.rowy.io/roadmap
keywords: Rowy, Pro, Plan, Airtable, Realtime, Continuous, Sync

Hi,
I just watched and liked [Airtable to Firebase - Scalable Airtable without row limits using Rowy](https://www.youtube.com/watch?v=zZxK9m2O_LI) but I'm still wondering if "Airtable realtime continuous data sync" is already released to PRO paid Plan : I don't see anything about it in https://roadmap.rowy.io/roadmap.
If it is Beta or Released, where can I find how it works and how to use it in [such](https://www.youtube.com/watch?v=zZxK9m2O_LI) a practical way !

Please note that I'm not yet PRO Plan but if that way of sync is in progress on roadmap, or in Beta, or released,
I will hurry to choose PRO Plan !

Best regards,

olπ

**Comment by BeeBombshell** [comment]
Hey @Olmatro, Airtable Import is free for all Rowy users. Here's some documentation on how to import/sync your Airtable data into Rowy: https://docs.rowy.io/import-export-data/import-airtable. Hope this helps! :)

**Reply by Olmatro** [reply]
Hi @BeeBombshell , thank you for your answer.

I just noticed I didn't provided your [link](https://docs.rowy.io/import-export-data/import-airtable) in my first message, but I played this just after watching [youtube Airtable Import rowy's video](https://www.youtube.com/watch?v=zZxK9m2O_LI).

I'm afraid Airtable is now providing noWay 
![Screen Shot 2023-02-28 at 12 53 54](https://user-images.githubusercontent.com/13352905/221845469-900142ef-be33-4ced-95d6-a59deec79651.png)
 row.assets[0].url since november 2022 or 09 february 2023 for those who asked Airtable for it , [one or two hour after Attachment has been created and forever](https://community.airtable.com/t5/show-tell/soon-you-ll-need-a-cdn/td-p/30799?lightbox-message-images-30799=1386iCC8EC1B4AD8094F9) explained and discussed [here](https://community.airtable.com/t5/show-tell/soon-you-ll-need-a-cdn/td-p/30799).

The result after evaluating Derivative Field is as painful as this:
![Screen Shot 2023-02-28 at 12 21 42](https://user-images.githubusercontent.com/13352905/221838157-b6aae7b5-74bb-4282-9203-83357d1526e1.png)

Any help is welcome :)

I'm supposing [GLIDE](https://www.glideapps.com/) got, and maybe payed airtable.com, for some non-public Airtable professional API, because [GLIDE](https://www.glideapps.com/) can do it from my Airtable Attachment Fields that contains valid picture file.

But [GLIDE](https://www.glideapps.com/) is NOT the solution for my Rowy's most frequent useCases: I need to successfully  https://docs.rowy.io/import-export-data/import-airtable anytime, and I probably need to trigger this to automate Derivative evaluation when some row.assets[0].url change occurs since last Derivative field evaluation.

oLπ

**Comment by BeeBombshell** [comment]
@Olmatro, once you import your data (including your JSON image field) into Rowy, the Derivative function that is set up to generate images would work. Let me explain why... 
The Derivative function extracts the image from the Airtable URL (which is supposed to expire after a few hours) and saves it in the Firebase Storage. This way, the image is now stored in the Firebase Store and the Airtable URL is not required anymore. From here, the Derivative function extracts the Firebase URL for your image (as in the screenshot below).

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/53828745/222377762-f74d732b-0b32-4ec0-bb46-12af6a6e2382.png">

**Reply by BeeBombshell** [reply]
As far as the error in the screenshot you attached is concerned, it is probably some other error (might as well be a bug). Could you try reloading the page to see if it works? Also, could you send the entire `ROWY TRACING ID` so that we can investigate what might be causing this?

**Reply by Olmatro** [reply]
Hi BeeBombshell,

Many thanks to follow this thread and to answer.
I suppose you are ROWY.io Collaborator :)

You would find the complete tracing ID here:
![Screen Shot 2023-02-26 at 23 12 13](https://user-images.githubusercontent.com/13352905/222431741-29dcadc5-8319-4d96-bfc5-0214cab50274.png)

I hope you will find there what was running wrong.

I will start another try & catch error (if occured) session from airtable and rowy.

Best regards,

olπ

**Reply by BeeBombshell** [reply]
@Olmatro, haha yes. :) 
I checked, and it seems to be a type error related to `downloadURL`, which is being used in your Derivative Script. Could you share the script you're using for your Derivative function?

**Comment by Olmatro** [comment]
Sorry I noticed I deleted accidentally the DERIVATIVE Column :(
But I rebuild it from the same ROWY's VIDEO:

![Airtable-to-Firebase-Scalable-Airtable-without-row-limits-using-Rowy-2023-02-25-at-15_00_16](https://user-images.githubusercontent.com/13352905/222516248-19c4dc07-cbbe-4742-b745-f802918f8735.png)

And Here's my DERIVATIVE Code:

const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
  const file = await rowy.storage.upload.url(row.thumbnail[0].url)
  return [file];
}

And Here's the RESULT:

![Screen Shot 2023-03-02 at 19 13 47](https://user-images.githubusercontent.com/13352905/222516609-059c85ee-655b-475b-9636-3290aabf6386.png)

If you would , I can share my Airtable with you, based on your mail.
My mail is jorganise.je partage@gmail.com

Best Regards,

olπ

**Comment by harinij** [comment, marked as answer]
@Olmatro It appears that you are using `row.thumbnail[0].url` which is invalid field. 

If you have the URL from Airtable in a column then you can write a code like this to upload the image to Firebase storage, where `airtableURL` is an URL from your Airtable import.
```
const file = await rowy.storage.upload.url(row.airtableURL)
return [file];
```

If you have an image column that you want to upload to Firestore then you use as following, as per the code you have but instead of `url` it will be `downloadURL` - because you need to pass the URL value to the upload function.

```
const file = await rowy.storage.upload.url(row.image[0].downloadURL)
return [file];
```

Let us know if that helps resolve your issue.

**Reply by Olmatro** [reply]
Hi Harini,
I especially thank you for personally intervening in this message thread.

I have already sent you [this link](url) by email this morning (CET) because it seems to me that this sequence of screenshots shows that in the case discussed here
it's AIRTABLE that has put [its warnings](https://community.airtable.com/t5/show-tell/soon-you-ll-need-a-cdn/td-p/30799) into action: after x hours or days or weeks, this link (url) doesn't reach any picture anywhere anymore.

AIRTABLE seems to apply [this recent measure](https://community.airtable.com/t5/show-tell/soon-you-ll-need-a-cdn/td-p/30799) very inconsistently and in most of the cases tested here,
AIRTABLE doesn't do it: the URL still allows to open the image and upload it to firebase via ROWY
or in a CMS underlying Glide.com.

I recently created an AIRTABLE Base populated with splash or public images and lorem ipsum to be able to develop and test my scripts and for the moment the problem doesn't happen again in any other BASE / Tables, neither after 2 hours nor after two days.

But I can't trust AIRTABLE about [this point](https://community.airtable.com/t5/show-tell/soon-you-ll-need-a-cdn/td-p/30799) that they decided not to serve anymore. 
So I'm busy improving the system and that's why I asked for a delay to publish
and to answer questions: I want to proceed in a systematic way by following a protocol and documenting
each experiment: this takes me a bit of time at a time of the year when I have very little so I progress slowly.

I am going to share with you as a reader of this AIRTABLE Base created only to serve experiments and reports
**and I ask if you would be available to show me how to share my ROWY tables populated from my AIRTABLE
Tables to share with you which would give you access to my WEBHOOKS when there are some and to my DERIVATIVES to be able to see my scripts related to my experiments.**

I think that my experiments shareable with whoever will ask me will totally document my attempts and results
which will allow to meet other similar issues and create synergies between Developers.

Best,

oLπ
